core/trino-main/src/main/java/io/trino/sql/planner/plan/TableWriterNode.java

```diff
@@ -537,17 +601,20 @@ public String toString()
     {
         private final TableExecuteHandle executeHandle;
         private final Optional<TableHandle> sourceHandle;
-        private final SchemaTableName schemaTableName;
+        private final QualifiedObjectName fullTableName;
+        private final ConnectorTableMetadata tableMetadata;
```

Comment:
This should be `TableHandle ...`

Reply:
The `EXECUTE` command can only be run over an existing table. An existing table can (and should be) identified by a `TableHandle`.

By looking at the code I can see that the `Optional<TableHandle> sourceHandle` is assigned later on based on the `TableScanNode`'s feeding into the `ExecuteNode`. I'm not sure why does it have to be this way and whether the `Optional<TableHandle> sourceHandle` will ever be different from the `TableHandle` identifying the table for `Execute`. Maybe @losipiuk can help us here?

However even if the `Optional<TableHandle> sourceHandle` could be different we can still store the original `TableHandle` identifying the table for the `EXECUTE` command and make a decision whether to enable table writers based on that.

Storing `tableMetadata` in the `TableExecuteTarget` is not ideal, as `ConnectorTableMetadata` is not `JSON` serializable and the `TableExecuteTarget` has to be.

Reply:
Indeed what Radek wrote above is true. The underlying table handle may change through optimizers execution - currently mostly due predicate/otherstuff pushdown.
Storing the original table handle as well as the "final" one would be confusing IMO. Maybe we can change the code to just store the minimal piece of information we need. Can we just add "supportsReportingWrittenBytes" where needed?
I have just skimmed the code so I am not sure how feasible that is.

Reply:
Right now, there is exactly one case when we use `EXECUTE`. In that case the `sourceTable === targetTable`. 
If we add in the future the anoter type of `EXECUTE` operation, then, probably we will need to deciding on `scaleWriters` basing on `operation type` as well. For now, we can go with that - I think so.

> However even if the Optional<TableHandle> sourceHandle could be different we can still store the original TableHandle identifying the table for the EXECUTE command and make a decision whether to enable table writers based on that.

I agree and I rewrite the code to use `tableHandle`.